R/simulateHubbell.R

```diff
@@ -49,34 +51,34 @@
 #'
 #' rowData <- t(rowData)
 #'
-#' ExampleHubbell <- simulateHubbell(N = 8, M = 10, I = 1000, d = 50,
+#' ExampleHubbell <- simulateHubbell(n.species = 8, M = 10, I = 1000, d = 50,
 #'                                                         m = 0.02, tend = 100)
 #' rowData(ExampleHubbell) <- rowData
 #' colData(ExampleHubbell) <- colData
 #'
-#' @return \code{simulateHubbell} returns a \code{\link{SummarizedExperiment}}
-#' object
+#' @return \code{simulateHubbell} returns a \linkS4class{SummarizedExperiment}
+#' object containing matrix with species abundance as rows and
+#' time points as columns
 #'
 #' @references Rosindell, James et al. "The unified neutral theory of
 #' biodiversity and biogeography at age ten." Trends in ecology & evolution
 #' vol. 26,7 (2011).
 #
 #' @export
 
-setGeneric("simulateHubbell",signature = "N",
-    function(N, M, I, d, m, tskip = 0, tend, norm = FALSE)
+setGeneric("simulateHubbell",signature = "n.species",
+    function(n.species, M, I, d, m, tskip = 0, tend, norm = FALSE)
         standardGeneric("simulateHubbell"))
 
-setMethod("simulateHubbell", signature = c(N="numeric"),
-    function(N, M, I, d, m, tskip = 0 , tend, norm = FALSE){
-            # First setting the function arguments right
-            pbirth <- runif(N, min = 0, max = 1)
+setMethod("simulateHubbell", signature = c(n.species="numeric"),
+    function(n.species, M, I, d, m, tskip = 0, tend, norm = FALSE){
+            pbirth <- runif(n.species, min = 0, max = 1)
             pmigr <- runif(M, min = 0, max = 1)
-            if(length(pbirth)!=N | length(pmigr)!=M){
-                stop("Either length of pbirth vector does not match with N or
-                    length of pmigr vector does not match with M")
+            if(length(pbirth)!=n.species | length(pmigr)!=M){
+                stop("Either length of pbirth vector does not match with
+                n.species or length of pmigr vector does not match with M")
```

Comment:
@danielriosgarza Do you think that we can remove this if-stop statement from here? Because in the original code, pbirth and pmigr were arguments of the function but since I put them into function, the length of pbirth and pmigr have to be the same with n.species and M anyway. What do you think ?

Reply:
Sure. You can remove it

Reply:
Or you can do, as part of the PR update. In either case, resolve the point after making the change so we get issues closed.